ledbuffer.h

```diff
@@ -127,38 +127,34 @@ class LEDBuffer
 
 class LEDBufferManager
 {
-    std::vector<std::unique_ptr<LEDBuffer>> _aptrBuffers;   // The circular array of buffer ptrs
-    size_t _iHeadIndex;                                     // Head pointer index
-    size_t _iTailIndex;                                     // Tail pointer index
-    size_t _cMaxBuffers;                                    // Number of buffers
-    mutable std::recursive_mutex _mutex;                    // Recursive mutex to protect the buffer
+    std::deque<std::unique_ptr<LEDBuffer>> _buffers;  // The deque of buffer pointers
+    size_t _cMaxBuffers;                              // Maximum number of buffers
+    mutable std::mutex _mutex;              // Recursive mutex to protect the buffer
 
 public:
     explicit LEDBufferManager(size_t cBuffers)
-        : _aptrBuffers(cBuffers), _iHeadIndex(0), _iTailIndex(0), _cMaxBuffers(cBuffers)
+        : _cMaxBuffers(cBuffers)
     {}
 
     double AgeOfOldestBuffer() const
     {
-        std::lock_guard<std::recursive_mutex> lock(_mutex);  // Lock the mutex before accessing the buffer
+        std::lock_guard<std::mutex> lock(_mutex);  // Lock the mutex before accessing the buffer

```

Comment:
You could use C++17's CTAD:
```suggestion
        std::lock_guard lock(_mutex);  // Lock the mutex before accessing the buffer
```

Reply:
You could also use `std::scoped_lock` instead, which is a newer API feature. In this case, it wouldn't do any difference.